### DISCUSSION TITLE
Updating Eldoc echo area after jumping with consult-flymake

### DISCUSSION THREAD

**Discussion by ashlineldridge** [minad/consult#1115] [Q&A]
When calling commands like `flymake-goto-next-error`, the Eldoc echo area is updated when jumping to a Flymake error. Additionally, Eldoc has the `eldoc-add-command` and `eldoc-add-command-completions` functions for registering additional commands that should trigger an update of Eldoc. For example, I have added this line to my config:

```
(eldoc-add-command-completions "xref-find-" "xref-go-" "xref-goto-")
```

So that jumping with Xref causes Eldoc to update. However, when I try to do the same with `consult-flymake` it doesn't work (e.g. `(eldoc-add-command "consult-flymake")`). I have `eldoc-echo-area-prefer-doc-buffer` set to `t` and if Eldoc buffer is visible _then_ `consult-flymake` will trigger an update of the Eldoc buffer. But if the Eldoc buffer is not visible, I can't seem to get `consult-flymake` to update the Eldoc echo area like I can with other commands. I have also tried using `:after` advice but didn't have any luck there either.

Any tips?

**Comment by minad** [comment]
Hmm, I haven't tried, but I suspect that the Consult preview is the problem since the buffer is updated while the completion command is still running. Maybe it is possible to trigger Eldoc somehow via the `consult-after-jump-hook`.
